src/app/component/user/components/profile/shopping-list/shopping-list.component.ts

```diff
@@ -1,19 +1,50 @@
+import { ProfileService } from './../profile-service/profile.service';
 import { Component, OnInit } from '@angular/core';
+import { ShoppingList } from '@global-user/models/shoppinglist.model';
 
 @Component({
   selector: 'app-shopping-list',
   templateUrl: './shopping-list.component.html',
   styleUrls: ['./shopping-list.component.scss']
 })
 export class ShoppingListComponent implements OnInit {
-  public shoppingList: Array<string> = ['Шоппер', 'Еко чашка', 'Бамбукова щітка'];
+  public shoppingList;
+  public profileSubscription;
 
-  constructor() { }
+  constructor(private profileService: ProfileService) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.getShoppingList();
+  }
+
+  public getShoppingList(): void {
+    this.profileSubscription = this.profileService.getShoppingList().subscribe(
+      (success: ShoppingList[]) => this.shoppingList = success,
+    );
+  }
 
-  public getShoppingList(): Array<string> {
-    return this.shoppingList.slice(0, 3);
+  public toggleDone(item): void {
+    this.profileService.toggleStatusOfShoppingItem(item)
+      .subscribe(
+        (success) => this.updateDataOnUi(item),
```

Comment:
If you have only one assignment or function call, you shouldn't write it in separate rows. Please, rewrite it and check the punctuation

Reply:
done